src/Ocelot/Request/Mapper/StreamHttpContent.cs

```diff
@@ -0,0 +1,113 @@
+﻿using Microsoft.AspNetCore.Http;
+using System.Buffers;
+
+namespace Ocelot.Request.Mapper;
+
+public class StreamHttpContent : HttpContent
+{
+    private const int DefaultBufferSize = 65536;
+    public const long UnknownLength = -1;
+    private readonly HttpContext _context;
+
+    public StreamHttpContent(HttpContext context)
+    {
+        _context = context ?? throw new ArgumentNullException(nameof(context));
+    }
+
+    protected override async Task SerializeToStreamAsync(Stream stream, TransportContext context,
+        CancellationToken cancellationToken)
+        => await CopyAsync(_context.Request.Body, stream, Headers.ContentLength ?? UnknownLength, false,
+            cancellationToken);
+
+    protected override async Task SerializeToStreamAsync(Stream stream, TransportContext context)
+        => await CopyAsync(_context.Request.Body, stream, Headers.ContentLength ?? UnknownLength, false,
+            CancellationToken.None);
+
+    protected override bool TryComputeLength(out long length)
+    {
+        length = -1;
+        return false;
+    }
+
+    // This is used internally by HttpContent.ReadAsStreamAsync(...)
+    protected override Task<Stream> CreateContentReadStreamAsync()
+    {
+        // Nobody should be calling this...
```

Comment:
@ggnaegi 
unfortunately, it can be called from a delegateHandler, like this for example
```
 // get upstream content
 if (request.Content != null)
 {
     await using var upstreamContentStream = await request.Content.ReadAsStreamAsync(cancellationToken);
     using var reader = new StreamReader(upstreamContentStream);
      ...
 }
```
Can't you leave the original?